### PR TITLE
[M1][xcodeproj] CLI to update xcodeproj rules tests

### LINKED ISSUE #598: Document steps to use rosetta to replicate github actions project stubs
Unless we drop the Intel github actions - You'll need to use rosetta to replicate our transitions / outputs as the CI does. It's not obvious how to replicate but rosetta 2 does a good job:
```
arch -arch x86_64 -c ~/Library/Caches/bazelisk/bin/bazelbuild/bazel-5.0.0-darwin-x86_64    run    tests/ios/xcodeproj:Test-With-Host-App-With-AdditionalPrebuildScript
```

When the community is fully on M1 laptops - it'll be good to have a way to recreate the intel stubs while github actions still has it

### PR DESCRIPTION
**Pull request by jerrymarino**
This PR adds a way for people to run tests which test rules_ios xcodeproj rules tests.  Anytime anyone looks at a change that impacts transitions ( including myself ) it causing the world more confusion to figure out the magic encantation of the test program and rosetta. You can easily update the test fixtures with `--update` if you just care to fix the transitions.

I added assertions for rosetta if you work on arm64 - now it just fails to prevent people getting stuck on this.

Update the fixtures with:
```
arch -arch x86_64 /bin/bash -l -c ./tests/xcodeproj-tests.sh --update
```

Fixes #598

Tested and working end to end on the latest cut of macOS Montery and Xcode 14.1.

It needs a login shell or `xcodebuild` doesn't work.

### CODE REVIEW

**Comment by jerrymarino** [comment]
After running this a few times over, I think it's probably fine to recurse into the login shell.

This might have some edge cases - but overall should work. If not, we can revert back to the warning...

**Review by justinseanmartin — APPROVED** [review]
Would it make sense to include a Bazel test rule for running the xcodeproj tests? It'd be nice to include an section in the repo readme about running the tests. It seems like it'd be nice if we could run `bazel test //...`. What do you think?

Also, another thought I had is we could go in a different direction and run the script through a sanitizer so they can match regardless of which machine architecture they're run on? That'd reduce the number of kinks in the requirements for running them, and I think still maintain a majority of what we care to be able to verify.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -137,7 +137,7 @@ jobs:
       - name: Select Xcode 14.2.0
         run: .github/workflows/xcode_select.sh
       - name: Run tests
-        run: ./tests/xcodeproj-tests.sh --clean && ./tests/test-tests.sh
+        run: ./tests/xcodeproj-tests.sh && ./tests/test-tests.sh
```
[the remark this answers is not in the document]
Comment (jerrymarino) [reply .github/workflows/tests.yml:140]: @justinseanmartin ah that's right good catch 👍

**Comment by jerrymarino** [comment]
> If we're not using --clean on CI anymore is this even necessary? Is it ok to change this behaviour on CI and remove the --clean flag? Are we testing the same thing as before if we remove it?

AFAIK it looks like the original code had a few programming problems that effectively didn't work. The if statements were looking like this:
```
if [[ CLEAN = 1 ]]; then
.. do stuf
fi
```

I like keeping the idea to clean them distinct from update. It's effectively 2 seperte modes of operations:
1. you have an existing proj 
2. you have a no existing proj

**Comment by thiagohmcruz** [comment]
> > I ran arch -arch x86_64 /bin/bash -l -c ./tests/xcodeproj-tests.sh --update as described in your PR description and the script just exists with this msg:
> 
> Well what was the issue and resolution you had? The default mode is the script will fail IME - if you don't have `bazelisk` installed?

The issue was gone after I fixed my `bazelisk` installation nvm!

**Review by thiagohmcruz — APPROVED** [review]

**Comment by jerrymarino** [comment]
TBH I don't think the program should call `bazelisk` anyhow. In the modern `bazelisk` program it will symlink it for you..

Perhaps someone can followup to rename `bazelisk` to `bazel` in this project and do tests on the shell that `bazel` works... 

```
bazel info 2>/dev/null || (echo "Bad bazel installation - fix this by installing a recent bazelisk or manually installing it" && exit 1)
```
